integration_tests/src/main/python/cache_test.py

```diff
@@ -234,3 +234,41 @@ def write_read_parquet_cached(spark):
           'spark.sql.parquet.outputTimestampType': ts_write}
 
     assert_gpu_and_cpu_are_equal_collect(read_parquet_cached(data_path_gpu), conf)
+
+@pytest.mark.parametrize('enableVectorized', ['false', 'true'], ids=idfn)
+@pytest.mark.parametrize('func', [with_cpu_session, with_gpu_session])
+@allow_non_gpu("ProjectExec", "Alias", "Literal", "DateAddInterval", "MakeInterval", "Cast",
+               "ExtractIntervalYears", "Year", "Month", "Second", "ExtractIntervalMonths",
+               "ExtractIntervalSeconds", "SecondWithFraction", "ColumnarToRowExec")
+@pytest.mark.parametrize('selectExpr', [("NULL as d", "d"),
```

Comment:
You have verified that this results in NullType being cached?

Reply:
Yes, its caching the NullType column as an IntegerType with null values.